databricks/koalas/series.py

```diff
@@ -65,7 +65,19 @@ def _init_from_spark(self, scol, kdf, metadata=None, *args):
     # arithmetic operators
     __neg__ = _column_op(spark.Column.__neg__)
     __add__ = _column_op(spark.Column.__add__)
-    __sub__ = _column_op(spark.Column.__sub__)
+
+    def __sub__(self, other):
+        assert isinstance(other, Series), "subtraction should only be applied to koalas.Series."
+
+        # Note that timestamp subtraction casts arguments to integer. This is to mimic Pandas's
+        # behaviours. Pandas returns 'timedelta64[ns]' from 'datetime64[ns]'s subtraction.
+        if isinstance(self.spark_type, TimestampType):
+            if not isinstance(other.spark_type, TimestampType):
+                raise TypeError('datetime subtraction can only be applied to datetime series.')
+            return Series(self._scol.cast('int') - other._scol.cast('int'), self._kdf)
```

Comment:
if astype already works, i'd prefer using astype so we minimize manual construction of Series.

Reply:
Looks it goes infinite loop .. :).